docs/DeveloperGuide.adoc

```diff
@@ -237,12 +243,68 @@ image::CommitActivityDiagram.png[]
 ** Cons: Requires dealing with commands that have already been undone: We must remember to skip these commands. Violates Single Responsibility Principle and Separation of Concerns as `HistoryManager` now needs to do two different things.
 // end::undoredo[]
 
-// tag::dataencryption[]
-=== [Proposed] Data Encryption
+// tag::annotation[]
+=== [Proposed] Annotation feature
+==== Proposed Implementation
+The annotation feature builds upon the structure of an offline document used to represent a _Readability-derived cache_.
+Such an offline document is represented by `OfflineDocument`, which contains at least one `Paragraph`. As each annotation is targeted to an entire paragraph,
+we can simply add an annotation to a `Paragraph`. The annotation will be internally stored as an `annotation`.
+You can view the class structure of the annotation feature in the Figure above.
+//TODO: link to annotation class diagram
+
+
+There are three parts to the annotation feature:
+ * Add an annotation (executed by `AddAnnotationCommand#execute()`).
+ * Delete an annotation (executed by `DeleteAnnotationCommand#execute()`).
+ * Edit an annotation (executed by `EditAnnotationCommand#execute()`).
+
+These executions are triggered by `annotation` command inputs entered into the `CommandBox`, which calls the respective command parser to create the command.
+Subsequently, the appropriate `Paragraph`(s) is retrieved and the annotation is handled according to the command.
+
+The following sequence diagram illustrates how the adding annotation operation works:
+
+image::AddAnnotationSequenceDiagram.png[]
+
+NOTE: The lifeline for `AddAnnotationCommandParser` and `AddAnnotationCommand` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+
+//TODO: after adding delete cmd, (highlight, note, both --> stray paragrph)
+For the delete operation, instead of creating and adding an `Annotation` to the retrieved `Paragraph`, the part or the entire pre-existing `Annotation` of the `Paragraph` will be dereferenced from the `Paragraph`.
+If the user requested to remove the entire annotation of the `Paragraph`, the `Annotation` is simply dereferenced and an `OfflineCommandResult` is returned.
+If the user requested to remove only the annotation note, the `AnnotationNote` of `Annotation` is dereferenced.
+Otherwise, if the user requested to remove only the highlight, the entire annotation is dereferenced from the `Paragraph` and added to a newly created `PhantomParagraph`.
```

Comment:
```suggestion
Otherwise, if the user requests to remove only the highlight, the entire annotation is dereferenced from the `Paragraph` and added to a newly created `PhantomParagraph`.
```